src/axom/quest/IntersectionShaper.hpp

```diff
@@ -829,91 +828,152 @@ class IntersectionShaper : public Shaper
 
-    SLIC_INFO(axom::fmt::format(axom::utilities::locale(),
-                                "Contour has been discretized into {:L} octahedra ",
-                                m_octcount));
+      axom::ArrayView<OctahedronType> octs_device_view = m_octs.view();
 
-    if(this->isVerbose())
-    {
-      // Print out the bounding box containing all the octahedra
-      BoundingBox3D all_oct_bb;
-      axom::Array<OctahedronType> octs_host = axom::Array<OctahedronType>(m_octs, host_allocator);
-      auto octs_host_view = octs_host.view();
+      AXOM_UNUSED_VAR(disc_status);  // silence warnings in release configs
+      SLIC_ASSERT_MSG(disc_status,
+                      "Discretization of contour has failed. Check that contour is valid");
 
-      for(int i = 0; i < m_octcount; i++)
+      SLIC_INFO(axom::fmt::format(axom::utilities::locale(),
+                                  "Contour has been discretized into {:L} octahedra ",
+                                  m_octcount));
+
+      if(this->isVerbose())
       {
-        all_oct_bb.addBox(primal::compute_bounding_box(octs_host[i]));
-      }
+        // Print out the bounding box containing all the octahedra
+        BoundingBox3D all_oct_bb;
+        axom::Array<OctahedronType> octs_host = axom::Array<OctahedronType>(m_octs, host_allocator);
+        auto octs_host_view = octs_host.view();
+
+        for(int i = 0; i < m_octcount; i++)
+        {
+          all_oct_bb.addBox(primal::compute_bounding_box(octs_host[i]));
+        }
+        SLIC_INFO(
+          axom::fmt::format("DEBUG: Bounding box containing all generated octahedra "
+                            "has dimensions:\n\t{}",
+                            all_oct_bb));
+
+        // Print out the total volume of all the octahedra
+        using REDUCE_POL = typename axom::execution_space<ExecSpace>::reduce_policy;
+        RAJA::ReduceSum<REDUCE_POL, double> total_oct_vol(0.0);
+        axom::for_all<ExecSpace>(
+          m_octcount,
+          AXOM_LAMBDA(axom::IndexType i) {
+            // Convert Octahedron into Polyhedrom
+            PolyhedronType octPoly;
+
+            octPoly.addVertex(octs_device_view[i][0]);
+            octPoly.addVertex(octs_device_view[i][1]);
+            octPoly.addVertex(octs_device_view[i][2]);
+            octPoly.addVertex(octs_device_view[i][3]);
+            octPoly.addVertex(octs_device_view[i][4]);
+            octPoly.addVertex(octs_device_view[i][5]);
+
+            octPoly.addNeighbors(0, {1, 5, 4, 2});
+            octPoly.addNeighbors(1, {0, 2, 3, 5});
+            octPoly.addNeighbors(2, {0, 4, 3, 1});
+            octPoly.addNeighbors(3, {1, 2, 4, 5});
+            octPoly.addNeighbors(4, {0, 5, 3, 2});
+            octPoly.addNeighbors(5, {0, 1, 3, 4});
+
+            total_oct_vol += octPoly.volume();
+          });
+
+        SLIC_INFO(axom::fmt::format("DEBUG: Total volume of all generated octahedra is {}",
```

Comment:
Would SLIC_DEBUG() be appropriate here?

Reply:
These Slic outputs are enabled when verbose flag is passed, so not really debug statements. Changed the output labels to "VERBOSE" to better fit what is happening here.